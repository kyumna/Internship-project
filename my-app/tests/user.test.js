import React from "react";
import { render } from "react-dom";
import Users from '../src/app/page'
import { act } from "react-dom/test-utils";
 const MOCK_DATA = [
  {
    "id": 1,
    "name": "Leanne Graham",
    "username": "Bret"
  },
];
let container;

// beforeEach function runs before each 
// of the tests in this file run to setup a testing environment  
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

// afterEach runs after each one of the tests in this file completes 
// to teardown the testing environment
afterEach(() => {
  document.body.removeChild(container);
  container = null;
});
describe("Posts", () => {
    it("Renders without crashing", async () => {
    
    // This part fakes fetch and resolves it by return MOCK_DATA
    global.fetch = jest.fn().mockImplementation(() =>
        Promise.resolve({
          json: () => Promise.resolve(MOCK_DATA),
        })
      );
      
      // This part renders a component inside of container that we created before
      await act(async () => {
        render(<Users />, container);
      });
  
      expect(container).toMatchSnapshot();
    });
  });