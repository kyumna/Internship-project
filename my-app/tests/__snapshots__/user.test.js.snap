// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Renders without crashing 1`] = `
<div>
  <div
    class="bg-pastelPink"
  >
    <a
      class="float-right mr-2 pt-2 font-bold"
      href="/Post"
    >
      See Posts
    </a>
    <h1
      class="text-center  text-2xl font-bold pb-2 pt-2 "
    >
      Users
    </h1>
    <div
      class="flex flex-wrap items-center justify-center h-screen "
    >
      <div
        class="w-64 h-64 p-2 m-4 border-1 border-gray-300 shadow-xl flex flex-wrap rounded-2xl cursor-pointer hover:transform hover:scale-110 transition-all duration-300"
      >
        <h2
          class=""
        >
          Leanne Graham
        </h2>
        <p
          class=""
        >
          Email: 
        </p>
        <p
          class=""
        >
          Phone: 
        </p>
      </div>
    </div>
  </div>
</div>
`;
